### PR TITLE
Add jobs resource

### PR DESCRIPTION
**Pull request by aripollak**
* Update Tableau Server API version to 3.1 to get jobs support
* Fix filtering of site names in cassettes
  * Regex in spec_helper.rb was expecting each site element to be on its own line, and it was allowing any entry that contained "group", which is contained by a key on every site element.

I kept getting a Bad Request response when trying to pass a job query filter in the query string to the jobs request and I didn't figure out why, but I left in support for that.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,16 +62,31 @@ docker exec -it CONTAINER_ID /bin/bash -c "cd /src && bundle && rake"
...
 
-Set the environment variables below to an administrator account, create a tunnel
-to your tableau server on port 2000, then run the commands below.
+To regenerate all the the cassettes, you'll first need to create the following on the Tableau server:
+* *Site*: Default
+* *Site*: TestSite
+ * *Datasource*: test
+ * *Username*: test_test
+
+And delete the following:
```
Comment (mattyb) [inline README.md:74]: delete the following **if they exist**?

File: lib/tableau_api/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module TableauApi
-  VERSION = '2.0.0'.freeze
+  VERSION = '2.1.0'.freeze
```
Comment (mattyb) [inline lib/tableau_api/version.rb:2]: Does enough break in the tableau api itself that we should make this a major version?

File: spec/resources/users_spec.rb [hunk 3]
```diff
@@ -6,12 +6,12 @@
   describe '#create' do
     # http://onlinehelp.tableau.com/v9.0/api/rest_api/en-us/help.htm#REST/rest_api_ref.htm#Add_User_to_Site%3FTocPath%3DAPI%2520Reference%7C_____7
     it 'can create a user in a site' do
-      user = client.users.create(username: 'test', site_role: 'Unlicensed')
+      user = client.users.create(username: 'test', site_role: 'ExplorerCanPublish')
```
Comment (mattyb) [inline spec/resources/users_spec.rb:9]: is this a tableau API change?

**Review by mattyb — CHANGES_REQUESTED** [review]
Just a couple questions!

File: lib/tableau_api/version.rb (hunk 2 above)
Comment (aripollak) [reply lib/tableau_api/version.rb:2]: Oh hmm yeah, because of the site roles.

File: spec/resources/users_spec.rb (hunk 3 above)
Comment (aripollak) [reply spec/resources/users_spec.rb:9]: Yeah, the roles are the one breaking change they said.

**Review by mattyb — APPROVED** [review]
LGTM!

**Pull request merged**